### DISCUSSION TITLE
Using PWM on ESP-01S

### DISCUSSION THREAD

**Discussion by mcmellow** [arendst/Tasmota#14351] [Q&A] [answered]
Dear all,

Can I use a EPS-01S for a PWM dimmer and use GPIO03 as a PWM pin?  The ESP-01S has only pin GPIO00 - GPIO03 and I want to use Tasmota on this DIY WiFi Dimmer.  https://github.com/mrWheel/DONOFF

Kind regards.

**Comment by sfromis** [comment, marked as answer]
Configuring gpio3 (RX) as PWM, to get varying LED brightness should work.

**Reply by mcmellow** [reply]
I am asking because in Restriction (https://tasmota.github.io/docs/Expanding-Tasmota/#restrictions) it is stated:
GPIO03 being RX is also good to avoid (PWM is not working on this GPIO).

**Reply by sfromis** [reply]
Yeah, that part was specifically what I answered, being careful not to imply that I commented on the whole dimmer.

**Reply by mcmellow** [reply]
> Yeah, that part was specifically what I answered, being careful not to imply that I commented on the whole dimmer.

Feel free to comment the whole dimmer. It all looks as a low cost DIY led dimmer with WiFi. (The PCB with all components for less then   € 7,00) 
Not using the original firmware, but trying with Tasmota and Domoticz.

**Reply by barbudor** [reply]
As said above, seeing the schematic I would definitively run away from that board. It's a killer.

What exactly do you want to dim ? Led ? bulb ? incandescent / halogen ? fluorescent (CFL) ? what voltage ? what current/power ?

**Reply by mcmellow** [reply]
Mains in NL 230V AC, using led bulbs 11W or less. 
The old fashion dimmers were OK for the OK for the old incandescent lamps. but not for the led bulbs. Now looking for WiFi or zigbee dimmers to control with Domoticz.  This dimmer will be build in a box ofcourse,

**Reply by sfromis** [reply]
> Feel free to comment the whole dimmer.

Barbudor is doing a much better job of that :-)

**Reply by barbudor** [reply]
As I said, the AC is rectified to 230 DC. 
A standard leb buld is unlikely to work because it includes a power supply that needs AC.

Get a proper dimmer. When life is at stake, 7€ is not the right price
Better get a Shelly Dimmer 2 or any chinese device listed in the [template](https://templates.blakadder.com/relay.html) will be safer that this thing

**Reply by mcmellow** [reply]
> As I said, the AC is rectified to 230 DC. A standard leb buld is unlikely to work because it includes a power supply that needs AC.
> 
That makes sence to me. Apart from the 230 DC on the output, I have to get DC Led bulbs as well.  Just bought all new AC versions. Thanks again for the warnings.

**Reply by mcmellow** [reply]
> Get a proper dimmer. When life is at stake, 7€ is not the right price Better get a Shelly Dimmer 2 or any chinese device listed in the [template](https://templates.blakadder.com/relay.html) will be safer that this thing

One last question and for my understanding. Do you know the technology Shelly is ussing for dimming  in their dimmers?

**Reply by barbudor** [reply]
Not sure as I havened openned it. I can do that later this week if this is of real interrest for you.
However what I can confirm is that it can support both Leading Edge and Trailing Edge 
So most probably a mosfet and not a triac

Let me know if you want me too look closer

**Reply by mcmellow** [reply]
Thanks for the answer.

**Comment by barbudor** [comment]
Beware that this device is working with AC rectified so the led output is at 220V DC
This is extremely dangerous high-voltage DC is much more dangerous than AC as the skin can be  "glued" to the wire making impossible to remove the contact.

**Reply by mcmellow** [reply]
Thanks for the warning.
